### PR TITLE
fix(@desktop/community): add online status to invite modal

### LINKED ISSUE #6985: Community invite modal - Contact online status is not shown
# Bug Report

## Description

In Figma contact online status is shown in invite list. This is not currently happening in the app. 

Found during testing of https://github.com/status-im/status-desktop/pull/6655 but raised separately due to it being dependant on a shared component being worked on separately - Confirmed with @MishkaRogachev 

https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=2563%3A524846

## Steps to reproduce

Invite a contact to community. User online status should be display on selection and confirmation stages.

#### Expected behavior
image

#### Actual behavior
image

### Additional Information

- Status desktop version: https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-6655/2/
- Operating System: Mac

### PR DESCRIPTION
**Pull request by faust4exe**
Fixes #6985

### What does the PR do

Add online status indication to Community invite modal.
Replace Contact component with StatusMemberListItem.
Add missing Nim functions to fill models with onlineStatus.
Adjust components paddings to match design. 

### Affected areas

Desktop Community

### Screenshot of functionality (including design for comparison)

- [X] I've checked the design and this PR matches it

Screenshot from 2022-08-24 12-33-22

Screenshot from 2022-08-24 13-50-03

https://user-images.githubusercontent.com/6445843/186401891-3f70efb6-0305-4a3f-a43d-aa6979c40e7d.mp4

### CODE REVIEW

File: ui/imports/shared/views/ExistingContacts.qml [hunk 1]
```diff
@@ -38,20 +41,32 @@ Item {
...
+        leftMargin: 0
+        spacing: Style.current.padding
 
         model: root.contactsStore.myContactsModel
-        delegate: Contact {
+        delegate: StatusMemberListItem {
             width: contactListView.availableWidth
-            showCheckbox: root.showCheckbox
-            isChecked: root.pubKeys.indexOf(model.pubKey) > -1
             pubKey: model.pubKey
```
Comment (osmaczko) [inline ui/imports/shared/views/ExistingContacts.qml:50]:
```suggestion
            pubKey: Utils.getCompressedPk(model.pubKey)
```

File: ui/imports/shared/views/PickedContacts.qml [hunk 2]
```diff
@@ -36,21 +38,32 @@ Item {
...
             filters: [
                 ExpressionFilter { expression: root.pubKeys.indexOf(model.pubKey) > -1 }
             ]
         }
-        delegate: Contact {
+
+        delegate: StatusMemberListItem {
             width: contactGridView.cellWidth
-            showCheckbox: false
             pubKey: model.pubKey
```
Comment (osmaczko) [inline ui/imports/shared/views/PickedContacts.qml:52]:
```suggestion
            pubKey: Utils.getCompressedPk(model.pubKey)
```

**Review by osmaczko — APPROVED** [review]

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml [hunk 3]
```diff
@@ -69,7 +71,7 @@ StatusStackModal {
         font.pixelSize: 13
         color: !!root.validationError ? Style.current.danger : Style.current.success
         Layout.alignment: Qt.AlignHCenter
```
[the remark this answers is not in the document]
Comment (faust4exe) [reply ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:73]: You mean the `subHeaderItem` ? Not sure why is should be removed, looks like functional, but I agree that Design does not contain it.

File: ui/imports/shared/views/ExistingContacts.qml (hunk 1 above)
Comment (faust4exe) [reply ui/imports/shared/views/ExistingContacts.qml:50]: So Design is wrong about visual format of pubKey ?

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (osmaczko) [reply ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:73]: I meant this line: `Layout.alignment: Qt.AlignHCenter`

File: ui/imports/shared/views/ExistingContacts.qml (hunk 1 above)
Comment (osmaczko) [reply ui/imports/shared/views/ExistingContacts.qml:50]: Yes, we should use compressed pubkey form everywhere.

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (faust4exe) [reply ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:73]: It also does not appear in Design, so I am not sure I should change it in any way, Or you have some other info about it ?

**Review by jrainville — COMMENTED** [review]
Good job, just address Patryk's comments first please.

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml (hunk 3 above)
[the remark this answers is not in the document]
Comment (faust4exe) [reply ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:73]: Oh - now it is more clear. Yes - you are correct it is not in Layout, so Layout.alighment can be removed , but looks like we need to replace it with anchor ?

[the remark this answers is not in the document]
Comment (faust4exe) [reply ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:73]: Or actually just `horizontalAlignment: Text.AlignHCenter`

**Review by osmaczko — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

File: ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml [hunk 4]
```diff
@@ -32,14 +33,27 @@ ColumnLayout {
 
     StatusInput {
         id: messageInput
-        label: qsTr("Invitation Message")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml:35]: I believe the former is correct; titles and headers should always be in title case, despite what the design shows

File: ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml [hunk 5]
```diff
@@ -56,7 +58,7 @@ StatusStackModal {
 
     finishButton: StatusButton {
         enabled: root.pubKeys.length > 0
-        text: qsTr("Send Invites")
+        text: qsTr("Send invites")
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/popups/community/InviteFriendsToCommunityPopup.qml:61]: Same here, keep it

**Review by caybro — CHANGES_REQUESTED** [review]

File: ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml (hunk 4 above)
Comment (faust4exe) [reply ui/app/AppLayouts/Chat/panels/communities/CommunityProfilePopupInviteMessagePanel.qml]: Reverted

**Comment by anastasiyaig** [comment]
okay it works as the issue says (fixed as expected) however I noticed that the indicator itself (green / grey) is not always reflecting the state correctly. I will investigate it in master

**Review by anastasiyaig — APPROVED** [review]

**Comment by faust4exe** [comment]
> okay it works as the issue says (fixed as expected) however I noticed that the indicator itself (green / grey) is not always reflecting the state correctly. I will investigate it in master

@anastasiyaig Yes - when I was testing I also observed that online status does not match one from Group with same users. But in this PR I just used Nim functions which already exist. So looks like we will require additional Nim fixes to make sure they work as expeced (but just in case I remember I saw online status once, and someone mentioned this status is updated every 5 minutes)

**Comment by anastasiyaig** [comment]
@faust4exe yes yes, it is fine, the status update is weirdo and i know it is not related to the changes u did in this PR :)

**Review by caybro — APPROVED** [review]

**Pull request merged**